### PR TITLE
Pass epoch argument to Comet Logger

### LINKED ISSUE #3437: Cometml Logger epoch is not set.
## 🐛 Bug

While logging using comet ml there is an argument to set epoch https://www.comet.ml/docs/python-sdk/Experiment/#experimentlog_metrics
The info is available in metrics dict, but instead of passing it as an arg, it is passed as metrics value. I will suply a PR in a moment

### PR DESCRIPTION
**Pull request by Vozf**
## What does this PR do?

Fixes #3437

CometLogger handles the epoch plotting differently. We need to pass it as an argument to the log_metrics method instead of as part of the dictionary.

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Vozf** [comment]
Well, seems like tests need that epoch key in metrics and modifying the dict is not an option. That way we can either
1. Create new dict for metrics before passing it to log_metrics
2. Pass both metrics with epoch key and epoch as an argument which is kind of strange duplicating
3. modifying the tests to not require epoch, that does not seem like an option

File: pytorch_lightning/loggers/comet.py [hunk 1]
```diff
@@ -229,7 +229,8 @@ def log_metrics(
             if is_tensor(val):
                 metrics[key] = val.cpu().detach()
 
-        self.experiment.log_metrics(metrics, step=step)
+        epoch = metrics.pop('epoch', None)
```
Comment (awaelchli) [inline pytorch_lightning/loggers/comet.py:232]: could you explain why pop and not get?

Comment (Vozf) [reply pytorch_lightning/loggers/comet.py:232]: I think that sending epoch value as a metric is kinda strange, because it is not. Moreover if we send both epoch as an argument and epoch as metric value they will be duplicated on the comet dashboard.
Although changing this to get will do the trick

Comment (awaelchli) [reply pytorch_lightning/loggers/comet.py:232]: Lightning plots epoch against global step in every logger, it's a good thing to have, because you will know how many steps one epoch corresponds to, especially in cases where epochs have different size.
Then, in your logging UI, you can choose epoch as the abscissa in your plots, since it is monotonic.

Comment (awaelchli) [reply pytorch_lightning/loggers/comet.py:232]: if you don't want it in comet, make a copy of the metrics dict, and pop it. maybe that works.

Comment (awaelchli) [reply pytorch_lightning/loggers/comet.py:232]: if comet has a special keyword for epoch already, maybe the solution is to copy the dict, pop epoch and pass the copy to the logger.

Comment (Vozf) [reply pytorch_lightning/loggers/comet.py:232]: Ok, got your point about other loggers. As you've said comet can handle epoch separately. I'll do the copying way

**Comment by Vozf** [comment]
@awaelchli is anything else required from my side? The only non succeded build is canceled somehow

**Comment by awaelchli** [comment]
@Vozf I think it's good. I wanted to add a test but didn't have time so far. Will check get back to you this evening

**Comment by awaelchli** [comment]
@Vozf Could you add this test in `tests/loggers/test_comet.py`?

```python 
def test_comet_epoch_logging(tmpdir, monkeypatch):
    import atexit
    monkeypatch.setattr(atexit, "register", lambda _: None)
    with patch("pytorch_lightning.loggers.comet.CometOfflineExperiment.log_metrics") as log_metrics:
        logger = CometLogger(project_name="test", save_dir=tmpdir)
        logger.log_metrics({"test": 1, "epoch": 1}, step=123)
        log_metrics.assert_called_once_with({"test": 1}, epoch=1, step=123)
```

File: tests/loggers/test_comet.py [hunk 2]
```diff
@@ -105,3 +105,12 @@ def test_comet_logger_dirs_creation(tmpdir, monkeypatch):
 
     assert trainer.checkpoint_callback.dirpath == (tmpdir / 'test' / version / 'checkpoints')
     assert set(os.listdir(trainer.checkpoint_callback.dirpath)) == {'epoch=0.ckpt'}
+
+
+def test_comet_epoch_logging(tmpdir, monkeypatch):
+    import atexit
```
Comment (awaelchli) [inline tests/loggers/test_comet.py:111]:
```suggestion
def test_comet_epoch_logging(tmpdir, monkeypatch):
    """ Test that CometLogger removes the epoch key from the metrics dict and passes it as argument. """
    import atexit
```

**Review by awaelchli — APPROVED** [review]

**Comment by Vozf** [comment]
Well, now even more tests are canceled. Is there a timout?

**Comment by awaelchli** [comment]
@Vozf yes, there is a timeout, this problem comes from master branch. your changes are fine, comet logger test passes.

**Comment by rohitgr7** [comment]
Does it create any difference if it is passed separately and not in metrics in the logs??

**Comment by Vozf** [comment]
Yes, the pre-defined x-axis `epoch` starts working.
Also no metrics made `epoch` chart is displayed and `current epoch` stat is displayed

**Comment by awaelchli** [comment]
so the comet logger has it's own way of tracking the epoch through the extra arg, right? they just rename it to current epoch?

**Comment by Vozf** [comment]
Yeah, they handle it separately. They rename it only in the main dashboard. In other places it is epoch as usual.

**Review by rohitgr7 — APPROVED** [review]
ok great :+1: tested. Not sure why comet-ml do it this way and not allow anything else on the x-axis except some selected parameters.

**Comment by Vozf** [comment]
Comet allows it, but to do so you need to manually create a chart select axis, smoothing, outliers and so on. 
But with epoch passed as an arg it automatically creates charts with x-axis epoch for all metrics.

**Comment by rohitgr7** [comment]
I tried to create one but it only allows to select the parameter for y-axis.. for the x-axis I can see only some specific parameter it shows to me.

**Comment by Vozf** [comment]
You are probably trying to do it in the charts section. But you need to go to the panel section to create chart with custom x axis

**Comment by rohitgr7** [comment]
Oh ok yeah, not familiar with comet completely :+1:

**Review by SkafteNicki — APPROVED** [review]

**Comment by Vozf** [comment]
Is anything else required for merge?
